File: packages/volume/ios/Plugin/Classes/Events/VolumeButtonPressedEvent.swift
import Foundation
import Capacitor

@objc public class VolumeButtonPressedEvent: NSObject, Result {
    let direction: String

    init(direction: String) {
        self.direction = direction
    }

    @objc public func toJSObject() -> AnyObject {
        var result = JSObject()
        result["direction"] = direction
        return result as AnyObject
    }
}

File: packages/volume/android/src/main/java/io/capawesome/capacitorjs/plugins/volume/classes/events/VolumeButtonPressedEvent.java
package io.capawesome.capacitorjs.plugins.volume.classes.events;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.volume.interfaces.Result;

public class VolumeButtonPressedEvent implements Result {

    @NonNull
    private final String direction;

    public VolumeButtonPressedEvent(@NonNull String direction) {
        this.direction = direction;
    }

    @Override
    @NonNull
    public JSObject toJSObject() {
        JSObject result = new JSObject();
        result.put("direction", direction);
        return result;
    }
}
